### ISSUE TITLE
Open fails to load

### ISSUE THREAD

**Issue by anonrig** [kdex/ws-promise-client#2]
I'm using Node 4.5 without Babel.

```javascript
'use strict';

const ws = require('ws-promise-client');
const debug = require('debug')('slack:websocket');

class Slack {
  constructor(url) {
    debug('Initialized to connect', url);
    this.url = url;

    this.connect()
      .then((response) => {
        debug('Connection established to Slack servers.');
        console.log('response', 'girdi');
      })
  }

  connect() {
    this.websocket = new ws.default(this.url);

    return this.websocket.open();
  }
}

module.exports = Slack;
```

My example code is like this. But the console.log doesn't get triggered. Also, the process gets ended without waiting for the open function's promise

**Comment by kdex** [comment]
First off, you've started dealing with asynchronous code, as `Slack#connect` returns a `Promise`. In Node, if `Promise`s get rejected, you won't get notified by default, so you might lose error traces if you don't `.catch` errors.

Consider working with something like:

```js
for (const event of ["unhandledRejection", "uncaughtException"]) {
        process.on(event, error => {
                console.error(error);
        });
}
```
Next, it seems you've found a bug that was introduced when `ws-promise-client` started to use `ws` as a WebSocket library for node environments. This is fixed in `4.1.4`.

Can you `npm i -S ws-promise-client` and tell me if everything works?

**Comment by kdex** [comment]
Closing issue; feel free to reply if you encounter further problems.
